ministryofjustice/analytics-platform-helm-charts
# Add chart for NFS user home dir backup cronjob

## What

Adds cronjob and configmap to run `s3cmd sync` every hour to backup NFS home dirs to S3

File patterns excluded from backups are listed in `charts/nfs-backup/templates/configmap.yml`

## How to review
1. Checkout this branch and associated PRs in [analytics-platform-ops](https://github.com/ministryofjustice/analytics-platform-ops/pull/112) and [analytics-platform-config](https://github.com/ministryofjustice/analytics-platform-config/pull/14)
2. Apply Terraform changes from `analytics-platform-ops` repo to ensure the S3 bucket and IAM resources exist
3. `helm install nfs-backup -f ../analytics-platform-config/chart-env-config/ENV/nfs-backup.yml`
4. Confirm that the cronjob is installed with `kubectl get cronjob`
5. Confirm that the job successfully runs on the hour by checking logs with `kubectl logs $POD_NAME`
